src/libraries/System.Console/src/System/Console.cs

```diff
@@ -400,7 +400,21 @@ public static int WindowWidth
             [UnsupportedOSPlatform("tvos")]
             get { return ConsolePal.WindowWidth; }
             [SupportedOSPlatform("windows")]
-            set { ConsolePal.WindowWidth = value; }
+            [SupportedOSPlatform("linux")]
+            [SupportedOSPlatform("freebsd")]
+            [SupportedOSPlatform("osx")]
+            set
+            {
+                if (Console.IsOutputRedirected)
+                {
+                    throw new IOException(SR.InvalidOperation_SetWindowSize);
+                }
+                if (value <= 0)
+                {
+                    throw new ArgumentOutOfRangeException(nameof(value), value, SR.ArgumentOutOfRange_NeedPosNum);
```

Comment:
aside- we can delete
https://github.com/dotnet/runtime/blob/399c6dc66ecf89bda90a6070da194fcc639e15ef/src/libraries/System.Private.CoreLib/src/System/ArgumentOutOfRangeException.cs#L51-L53
https://github.com/dotnet/runtime/blob/399c6dc66ecf89bda90a6070da194fcc639e15ef/src/libraries/System.Private.CoreLib/src/System/ArgumentOutOfRangeException.cs#L90-L92

I dont know why we wouldn't use it in the libraries, as you are doing here, and indeed I see we are already doing in some places eg
https://github.com/dotnet/runtime/blob/399c6dc66ecf89bda90a6070da194fcc639e15ef/src/libraries/Common/src/System/IO/StreamHelpers.CopyValidation.cs#L19

It makes a nicer message.

Reply:
Thank you for the note.
These are where I've referred to
https://github.com/dotnet/runtime/blob/60be1deb89d1d811bd0380056d7831ba622d9409/src/libraries/System.Console/src/System/Console.cs#L547-L551
https://github.com/dotnet/runtime/blob/60be1deb89d1d811bd0380056d7831ba622d9409/src/libraries/System.Console/src/System/ConsolePal.Windows.cs#L672-L686